ai/src/main/java/com/alibaba/nacos/ai/index/McpCacheIndex.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 1999-2025 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacos.ai.index;
+
+import com.alibaba.nacos.ai.model.mcp.McpServerIndexData;
+
+/**
+ * MCP cache index interface providing fast mapping between MCP Name and MCP ID.
+ *
+ * @author misselvexu
+ */
+public interface McpCacheIndex {
```

Comment:
What's the diffenent with `McpServerIndex`?

Reply:
See specifically the response below, ["MCP Server Index Architecture - Core Relationships"](https://github.com/alibaba/nacos/pull/13604#issuecomment-3067710654).